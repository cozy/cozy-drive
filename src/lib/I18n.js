/**
 * preact plugin that provides Polyglot and date-fns helpers using a Higher Order Component.
 *
 */

'use strict'

import React, { Component } from 'react'
import Polyglot from 'node-polyglot'
import format from 'date-fns/format'
import en from '../locales/en'

const initPolyglot = (context, lang) => {
  const polyglot = new Polyglot({
    phrases: en,
    locale: 'en'
  })

  // Load global locales
  if (lang && lang !== 'en') {
    try {
      const dict = require(`../locales/${lang}`)
      polyglot.extend(dict)
      polyglot.locale(lang)
    } catch (e) {
      console.warn(`The dict phrases for "${lang}" can't be loaded`)
    }
  }

  // Load context locales
  if (context) {
    const dict = require(`../contexts/${context}/locales/${lang}`)
    polyglot.extend(dict)
  }

  return polyglot
}

const initFormat = lang => {
  const locales = {
    en: require('date-fns/locale/en')
  }
  if (lang && lang !== 'en') {
    try {
      locales[lang] = require(`date-fns/locale/${lang}`)
    } catch (e) {
      console.warn(`The "${lang}" locale isn't supported by date-fns`)
    }
  }
  return (date, formatStr) => format(date, formatStr, { locale: locales[lang] })
}

// Provider root component
export class I18n extends Component {
  constructor (props) {
    super(props)
    this.polyglot = initPolyglot(props.context, props.lang)
    this.format = initFormat(props.lang)
  }

  getChildContext () {
    return {
      t: this.polyglot.t.bind(this.polyglot),
      f: this.format
    }
  }

  componentWillReceiveProps (newProps) {
    if (newProps.locale !== this.props.locale) {
      this.polyglot = initPolyglot(newProps.context, newProps.lang)
      this.format = initFormat(newProps.lang)
    }
  }

  render () {
    return (this.props.children && this.props.children[0]) || null
  }
}

I18n.childContextTypes = {
  t: React.PropTypes.func,
  f: React.PropTypes.func
}

// higher order decorator for components that need `t` and/or `f`
export const translate = () => {
  return (WrappedComponent) => {
    const _translate = (props, context) => (
      <WrappedComponent {...props} t={context.t} f={context.f} />
    )
    return _translate
  }
}
