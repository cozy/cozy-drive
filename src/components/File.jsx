import React, { Component } from 'react'
import classNames from 'classnames'
import filesize from 'filesize'
import { withRouter } from 'react-router'
import Hammer from 'hammerjs'

import styles from '../styles/table'
import { translate } from 'cozy-ui/react/I18n'
import RenameInput from '../ducks/files/RenameInput'
import { isDirectory } from '../ducks/files/files'
import Spinner from 'cozy-ui/react/Spinner'

import { getFolderUrl } from '../reducers'

export const splitFilename = file => isDirectory(file)
  ? { filename: file.name, extension: '' }
  : {
    extension: file.name.slice(file.name.lastIndexOf('.') + 1),
    filename: file.name.slice(0, file.name.lastIndexOf('.') + 1)
  }

export const getClassFromMime = (attrs) => {
  if (isDirectory(attrs)) {
    return styles['fil-file-folder']
  }
  return styles['fil-file-' + attrs.mime.split('/')[0]] || styles['fil-file-files']
}

const getClassDiv = element => {
  if (element.nodeName === 'DIV') {
    return element.className
  }
  return getClassDiv(element.parentNode)
}

const enableTouchEvents = ev => {
  // remove event when you rename a file
  if (['INPUT', 'BUTTON'].indexOf(ev.target.nodeName) !== -1) {
    return false
  }

  // remove event when it's checkbox (it's already trigger, but Hammer don't respect stopPropagation)
  if (getClassDiv(ev.target).indexOf(styles['fil-content-file-select']) !== -1) {
    return false
  }

  return true
}

class File extends Component {
  constructor (props) {
    super(props)
    this.state = {
      opening: false
    }
  }

  componentDidMount () {
    this.gesturesHandler = new Hammer.Manager(this.fil)
    this.gesturesHandler.add(new Hammer.Tap({ event: 'singletap' }))
    this.gesturesHandler.add(new Hammer.Press({ event: 'onpress' }))
    this.gesturesHandler.on('onpress singletap', (ev) => {
      if (enableTouchEvents(ev)) {
        if (ev.type === 'onpress' || this.props.selectionModeActive) {
          this.toggle(ev.srcEvent)
        } else {
          this.open(ev.srcEvent, this.props.attributes)
        }
      }
    })
  }

  componentWillUnmount () {
    this.gesturesHandler.destroy()
  }

  toggle (e) {
    e.stopPropagation()
    const { attributes, onToggle, selected } = this.props
    onToggle(attributes, selected)
  }

  open (e, attributes) {
    e.stopPropagation()
    if (isDirectory(attributes)) {
      this.setState({ opening: true })
      this.props.onFolderOpen(attributes.id).then(() => {
        this.setState({ opening: false })
        this.props.router.push(getFolderUrl(attributes.id, this.props.location))
      })
    } else {
      this.props.onFileOpen(this.props.displayedFolder, attributes)
    }
  }

  render ({ t, f, style, attributes, selected, selectionModeActive, onShowActionMenu, isRenaming }, { opening }) {
    return (
      <div
        ref={fil => { this.fil = fil }}
        style={style}
        className={classNames(
          styles['fil-content-row'],
          selected ? styles['fil-content-row-selected'] : '',
          { [styles['fil-content-row--selectable']]: selectionModeActive }
        )}
      >
        <div className={classNames(styles['fil-content-cell'], styles['fil-content-file-select'])} onclick={(e) => this.toggle(e)}>
          <span data-input='checkbox'>
            <input
              type='checkbox'
              checked={selected}
             />
            <label />
          </span>
        </div>
        {this.renderFilenameCell(attributes, opening, isRenaming)}
        <div className={classNames(styles['fil-content-cell'], styles['fil-content-date'])}>
          <time datetime=''>{ f(attributes.created_at, t('table.row_update_format')) }</time>
        </div>
        <div className={classNames(styles['fil-content-cell'], styles['fil-content-size'])}>
          {isDirectory(attributes)
            ? '—'
            : filesize(attributes.size, {base: 10})}
        </div>
        <div className={classNames(styles['fil-content-cell'], styles['fil-content-status'])}>—</div>
        <div className={classNames(styles['fil-content-cell'], styles['fil-content-file-action'])}>
          <button className='coz-btn coz-btn--extra' onClick={(e) => {
            onShowActionMenu(attributes.id)
            e.stopPropagation()
          }} />
        </div>
      </div>
    )
  }

  renderFilenameCell (attributes, opening, isRenaming) {
    const classes = classNames(
      styles['fil-content-cell'],
      styles['fil-content-file'],
      getClassFromMime(attributes),
      { [styles['fil-content-file-openable']]: !isRenaming }
    )
    const { filename, extension } = splitFilename(attributes)
    return (
      <div className={classes}>
        {isRenaming
          ? <RenameInput />
          : <div>
            {filename}
            {extension && <span className={styles['fil-content-ext']}>{extension}</span>}
            {opening === true && <Spinner />}
          </div>
        }
      </div>
    )
  }
}

export default withRouter(translate()(File))

export const FilePlaceholder = ({ style }) => (
  <div
    style={style}
    className={styles['fil-content-row']}
  >
    <div className={classNames(styles['fil-content-cell'], styles['fil-content-file-select'])} />
    <div className={classNames(styles['fil-content-cell'], styles['fil-content-file'])}>
      <div className={styles['fil-content-file-placeholder']} />
    </div>
    <div className={classNames(styles['fil-content-cell'], styles['fil-content-date'])} />
    <div className={classNames(styles['fil-content-cell'], styles['fil-content-size'])} />
    <div className={classNames(styles['fil-content-cell'], styles['fil-content-status'])} />
    <div className={classNames(styles['fil-content-cell'], styles['fil-content-file-action'])} />
  </div>
)
