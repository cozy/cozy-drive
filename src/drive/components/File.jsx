/* global cozy */
import React, { Component } from 'react'
import classNames from 'classnames'
import filesize from 'filesize'
import { withRouter, Link } from 'react-router'
import Hammer from 'hammerjs'

import styles from '../styles/table'
import { translate } from 'cozy-ui/react/I18n'
import RenameInput from '../ducks/files/RenameInput'
import { isDirectory } from '../ducks/files/files'
import Spinner from 'cozy-ui/react/Spinner'
import Preview from '../components/Preview'
import breakpointsAware from 'cozy-ui/react/helpers/breakpoints'

import { getFolderUrl } from '../reducers'

export const splitFilename = file =>
  isDirectory(file)
    ? { filename: file.name, extension: '' }
    : {
        extension: file.name.slice(file.name.lastIndexOf('.') + 1),
        filename: file.name.slice(0, file.name.lastIndexOf('.') + 1)
      }

export const getClassFromMime = attrs => {
  if (isDirectory(attrs)) {
    return styles['fil-file-folder']
  }

  const [type, subtype] = attrs.mime.split('/')
  let className = 'files'

  if (styles['fil-file-' + type]) {
    className = type
  } else if (type === 'application') {
    // for mime types with an `application` type, we need to consider the subtype
    const correlationTable = {
      zip: 'zip',
      pdf: 'pdf',
      spreadsheet: 'sheet',
      excel: 'sheet',
      presentation: 'slide',
      powerpoint: 'slide'
    }

    const reg = new RegExp(Object.keys(correlationTable).join('|'))
    const result = subtype.match(reg)

    if (result) className = correlationTable[result[0]]
  }

  return styles['fil-file-' + className]
}

const getParentDiv = element => {
  if (element.nodeName.toLowerCase() === 'div') {
    return element
  }
  return getParentDiv(element.parentNode)
}

const enableTouchEvents = ev => {
  // remove event when you rename a file
  if (['INPUT', 'BUTTON'].indexOf(ev.target.nodeName) !== -1) {
    return false
  }

  // remove event when it's checkbox (it's already trigger, but Hammer don't respect stopPropagation)
  if (
    getParentDiv(ev.target).className.indexOf(
      styles['fil-content-file-select']
    ) !== -1
  ) {
    return false
  }

  // remove events when they are on the file's path, because it's a different behavior
  if (
    ev.target.nodeName.toLowerCase() === 'a' &&
    ev.target.className.indexOf(styles['fil-file-path']) >= 0
  ) {
    return false
  }

  return true
}

class File extends Component {
  constructor(props) {
    super(props)
    this.state = {
      opening: false
    }
  }

  componentDidMount() {
    this.gesturesHandler = new Hammer.Manager(this.fil)
    this.gesturesHandler.add(new Hammer.Tap({ event: 'singletap' }))
    this.gesturesHandler.add(new Hammer.Press({ event: 'onpress' }))
    this.gesturesHandler.on('onpress singletap', ev => {
      if (enableTouchEvents(ev)) {
        if (ev.type === 'onpress' || this.props.selectionModeActive) {
          this.toggle(ev.srcEvent)
        } else {
          this.open(ev.srcEvent, this.props.attributes)
        }
      }
    })
  }

  componentWillUnmount() {
    this.gesturesHandler.destroy()
  }

  toggle(e) {
    e.stopPropagation()
    const { attributes, onToggle, selected } = this.props
    onToggle(attributes, selected)
  }

  open(e, attributes) {
    e.stopPropagation()
    if (isDirectory(attributes)) {
      this.setState({ opening: true })
      this.props.onFolderOpen(attributes.id).then(() => {
        this.setState({ opening: false })
        this.props.router.push(getFolderUrl(attributes.id, this.props.location))
      })
    } else {
      this.props.onFileOpen({
        ...attributes,
        availableOffline: this.props.isAvailableOffline
      })
    }
  }

  render() {
    const {
      t,
      f,
      style,
      attributes,
      selected,
      selectionModeActive,
      onShowActionMenu,
      isRenaming,
      withSelectionCheckbox,
      withFilePath,
      isAvailableOffline,
      breakpoints: { isExtraLarge, isMobile }
    } = this.props
    const { opening } = this.state
    return (
      <div
        ref={fil => {
          this.fil = fil
        }}
        style={style}
        className={classNames(
          styles['fil-content-row'],
          selected ? styles['fil-content-row-selected'] : '',
          { [styles['fil-content-row--selectable']]: selectionModeActive }
        )}
      >
        <div
          className={classNames(
            styles['fil-content-cell'],
            styles['fil-content-file-select']
          )}
          onClick={e => this.toggle(e)}
        >
          {withSelectionCheckbox && (
            <span data-input="checkbox">
              <input type="checkbox" checked={selected} />
              <label />
            </span>
          )}
        </div>
        <FileNameCell
          attributes={attributes}
          isRenaming={isRenaming}
          opening={opening}
          withFilePath={withFilePath}
          isMobile={isMobile}
        />
        <div
          className={classNames(
            styles['fil-content-cell'],
            styles['fil-content-date']
          )}
        >
          <time dateTime={attributes.updated_at || attributes.created_at}>
            {f(
              attributes.updated_at || attributes.created_at,
              `${
                isExtraLarge
                  ? t('table.row_update_format_full')
                  : t('table.row_update_format')
              }`
            )}
          </time>
        </div>
        <div
          className={classNames(
            styles['fil-content-cell'],
            styles['fil-content-size']
          )}
        >
          {isDirectory(attributes)
            ? 'â€”'
            : filesize(attributes.size, { base: 10 })}
        </div>
        {isAvailableOffline && <AvailableOfflineBadge />}
        <div
          className={classNames(
            styles['fil-content-cell'],
            styles['fil-content-file-action']
          )}
        >
          <button
            className={classNames(styles['c-btn'], styles['c-btn-extra'])}
            onClick={e => {
              onShowActionMenu(attributes.id)
              e.stopPropagation()
            }}
          />
        </div>
      </div>
    )
  }
}

const AvailableOfflineBadge = props => (
  <div
    className={classNames(
      styles['fil-content-cell'],
      styles['fil-content-status']
    )}
  >
    <span className={styles['fil-content-offline']} />
  </div>
)

const FileNameCell = ({
  attributes,
  isRenaming,
  opening,
  withFilePath,
  isMobile
}) => {
  const classes = classNames(
    styles['fil-content-cell'],
    styles['fil-content-file'],
    getClassFromMime(attributes),
    { [styles['fil-content-file-openable']]: !isRenaming }
  )
  const { filename, extension } = splitFilename(attributes)
  const url = cozy.client._url
  return (
    <div className={classes}>
      {attributes.links &&
        attributes.links.small && (
          <Preview thumbnail={`${url}${attributes.links.small}`} />
        )}
      {isRenaming ? (
        <RenameInput />
      ) : (
        <div className={styles['fil-file']}>
          <div className={styles['fil-file-filename']}>
            {filename}
            {extension && (
              <span className={styles['fil-content-ext']}>{extension}</span>
            )}
            {opening === true && <Spinner />}
          </div>
          {withFilePath &&
            (isMobile ? (
              <div className={styles['fil-file-path']}>{attributes.path}</div>
            ) : (
              <Link
                to={`/folder/${attributes.dir_id}`}
                className={styles['fil-file-path']}
              >
                {attributes.path}
              </Link>
            ))}
        </div>
      )}
    </div>
  )
}

export default breakpointsAware()(withRouter(translate()(File)))

export const FilePlaceholder = ({ style }) => (
  <div style={style} className={styles['fil-content-row']}>
    <div
      className={classNames(
        styles['fil-content-cell'],
        styles['fil-content-file-select']
      )}
    />
    <div
      className={classNames(
        styles['fil-content-cell'],
        styles['fil-content-file']
      )}
    >
      <div className={styles['fil-content-file-placeholder']} />
    </div>
    <div
      className={classNames(
        styles['fil-content-cell'],
        styles['fil-content-date']
      )}
    />
    <div
      className={classNames(
        styles['fil-content-cell'],
        styles['fil-content-size']
      )}
    />
    <div
      className={classNames(
        styles['fil-content-cell'],
        styles['fil-content-status']
      )}
    />
    <div
      className={classNames(
        styles['fil-content-cell'],
        styles['fil-content-file-action']
      )}
    />
  </div>
)
