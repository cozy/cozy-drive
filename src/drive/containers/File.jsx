import React, { Component } from 'react'
import classNames from 'classnames'
import filesize from 'filesize'
import { withRouter, Link } from 'react-router'
import Hammer from 'hammerjs'

import styles from '../styles/table'
import { translate } from 'cozy-ui/react/I18n'
import RenameInput from '../ducks/files/RenameInput'
import { isDirectory } from '../ducks/files/files'
import Spinner from 'cozy-ui/react/Spinner'
import { ImageLoader } from 'components/Image'
import { Button, Icon, withBreakpoints, MidEllipsis } from 'cozy-ui/react'
import { SharedBadge, SharedStatus } from 'sharing'
import { getFileMimetype } from 'drive/lib/getFileMimetype'

import { getFolderUrl } from '../reducers'

export const splitFilename = file =>
  isDirectory(file)
    ? { filename: file.name, extension: '' }
    : {
        extension: file.name.slice(file.name.lastIndexOf('.')),
        filename: file.name.slice(0, file.name.lastIndexOf('.'))
      }

export const getClassFromMime = attrs => {
  if (isDirectory(attrs)) {
    return styles['fil-file-folder']
  }
  return styles[
    'fil-file-' +
      (getFileMimetype(styles, 'fil-file-')(attrs.mime, attrs.name) || 'files')
  ]
}

const getParentDiv = element => {
  if (element.nodeName.toLowerCase() === 'div') {
    return element
  }
  return getParentDiv(element.parentNode)
}

const enableTouchEvents = ev => {
  // remove event when you rename a file
  if (['INPUT', 'BUTTON'].indexOf(ev.target.nodeName) !== -1) {
    return false
  }

  const parentDiv = getParentDiv(ev.target)
  // remove event when it's the checkbox or the more button
  if (
    parentDiv.className.indexOf(styles['fil-content-file-select']) !== -1 ||
    parentDiv.className.indexOf(styles['fil-content-file-action']) !== -1
  ) {
    return false
  }

  // remove events when they are on the file's path, because it's a different behavior
  if (
    ev.target.nodeName.toLowerCase() === 'a' &&
    ev.target.className.indexOf(styles['fil-file-path']) >= 0
  ) {
    return false
  }

  return true
}

const SelectBox = ({ withSelectionCheckbox, selected, onClick }) => (
  <div
    className={classNames(
      styles['fil-content-cell'],
      styles['fil-content-file-select']
    )}
    onClick={onClick}
  >
    {withSelectionCheckbox && (
      <span data-input="checkbox">
        <input type="checkbox" checked={selected} />
        <label />
      </span>
    )}
  </div>
)

const FileName = ({
  attributes,
  isRenaming,
  opening,
  withFilePath,
  withSharedBadge,
  isMobile,
  formattedSize,
  formattedUpdatedAt
}) => {
  const classes = classNames(
    styles['fil-content-cell'],
    styles['fil-content-file'],
    getClassFromMime(attributes),
    { [styles['fil-content-file-openable']]: !isRenaming }
  )
  const { filename, extension } = splitFilename(attributes)
  return (
    <div className={classes}>
      {attributes.class === 'image' && (
        <ImageLoader
          file={attributes}
          size="small"
          render={src => (
            <div
              className={styles['fil-file-preview']}
              style={`background-image: url(${src});`}
            />
          )}
        />
      )}
      {withSharedBadge && (
        <SharedBadge
          docId={attributes.id}
          className={styles['fil-content-shared']}
          xsmall
        />
      )}
      {isRenaming ? (
        <RenameInput />
      ) : (
        <div className={styles['fil-file']}>
          <div className={styles['fil-file-filename']}>
            <div className={styles['fil-file-filename-wrapper']}>
              <div className={styles['fil-file-filename-and-ext']}>
                {filename}
                {extension && (
                  <span className={styles['fil-content-ext']}>{extension}</span>
                )}
              </div>
              <div className={styles['fil-file-filename-spinner']}>
                {opening === true && <Spinner />}
              </div>
            </div>
          </div>
          {withFilePath &&
            attributes.path &&
            (isMobile ? (
              <MidEllipsis
                className={styles['fil-file-path']}
                text={attributes.path}
              />
            ) : (
              <Link
                to={`/folder/${attributes.dir_id}`}
                className={styles['fil-file-path']}
              >
                <MidEllipsis text={attributes.path} />
              </Link>
            ))}
          {!withFilePath &&
            (isDirectory(attributes) || (
              <div className={styles['fil-file-infos']}>
                {`${formattedUpdatedAt}${
                  formattedSize ? ` - ${formattedSize}` : ''
                }`}
              </div>
            ))}
        </div>
      )}
    </div>
  )
}

const LastUpdate = ({ date, formatted = '-' }) => (
  <div
    className={classNames(
      styles['fil-content-cell'],
      styles['fil-content-date']
    )}
  >
    <time dateTime={date}>{formatted}</time>
  </div>
)

const Size = ({ filesize = '-' }) => (
  <div
    className={classNames(
      styles['fil-content-cell'],
      styles['fil-content-size']
    )}
  >
    {filesize}
  </div>
)

const Status = ({ isAvailableOffline, id }) => (
  <div
    className={classNames(
      styles['fil-content-cell'],
      styles['fil-content-status']
    )}
  >
    {isAvailableOffline && (
      <span className={styles['fil-content-offline']}>
        <Icon icon="phone-download" color="white" width="14" height="14" />
      </span>
    )}
    <SharedStatus docId={id} className={styles['fil-content-sharestatus']} />
  </div>
)

const FileAction = ({ onClick }) => (
  <div
    className={classNames(
      styles['fil-content-cell'],
      styles['fil-content-file-action']
    )}
  >
    <Button
      theme="action"
      onClick={onClick}
      extension="narrow"
      icon={<Icon icon="dots" color="charcoalGrey" width="17" height="17" />}
    />
  </div>
)

class File extends Component {
  state = {
    opening: false
  }

  componentDidMount() {
    this.gesturesHandler = new Hammer.Manager(this.filerow)
    this.gesturesHandler.add(new Hammer.Tap({ event: 'singletap' }))
    this.gesturesHandler.add(new Hammer.Press({ event: 'onpress' }))
    this.gesturesHandler.on('onpress singletap', ev => {
      if (enableTouchEvents(ev)) {
        if (ev.type === 'onpress' || this.props.selectionModeActive) {
          this.toggle(ev.srcEvent)
        } else {
          this.open(ev.srcEvent, this.props.attributes)
        }
      }
    })
  }

  componentWillUnmount() {
    this.gesturesHandler.destroy()
  }

  toggle(e) {
    e.stopPropagation()
    const { attributes, onToggle, selected } = this.props
    onToggle(attributes, selected)
  }

  open(e, attributes) {
    e.stopPropagation()
    if (isDirectory(attributes)) {
      this.setState(state => ({ ...state, opening: true }))
      this.props.onFolderOpen(attributes.id).then(() => {
        this.setState(state => ({ ...state, opening: false }))
        this.props.router.push(getFolderUrl(attributes.id, this.props.location))
      })
    } else {
      this.props.onFileOpen({
        ...attributes,
        availableOffline: this.props.isAvailableOffline
      })
    }
  }

  render() {
    const {
      t,
      f,
      attributes,
      selected,
      selectionModeActive,
      onShowActionMenu,
      isRenaming,
      withSelectionCheckbox,
      withFilePath,
      withSharedBadge,
      isAvailableOffline,
      breakpoints: { isExtraLarge, isMobile }
    } = this.props
    const { opening } = this.state
    const filContentRowSelected = classNames(
      styles['fil-content-row'],
      selected ? styles['fil-content-row-selected'] : '',
      { [styles['fil-content-row--selectable']]: selectionModeActive }
    )
    const formattedSize = isDirectory(attributes)
      ? undefined
      : filesize(attributes.size, { base: 10 })

    const updatedAt = attributes.updated_at || attributes.created_at
    const formattedUpdatedAt = f(
      updatedAt,
      isExtraLarge
        ? t('table.row_update_format_full')
        : t('table.row_update_format')
    )
    return (
      <div
        ref={filerow => {
          this.filerow = filerow
        }}
        className={filContentRowSelected}
      >
        <SelectBox
          withSelectionCheckbox={withSelectionCheckbox}
          selected={selected}
          onClick={e => this.toggle(e)}
        />
        <FileName
          attributes={attributes}
          isRenaming={isRenaming}
          opening={opening}
          withFilePath={withFilePath}
          withSharedBadge={withSharedBadge}
          isMobile={isMobile}
          formattedSize={formattedSize}
          formattedUpdatedAt={formattedUpdatedAt}
        />
        <LastUpdate
          date={updatedAt}
          formatted={isDirectory(attributes) ? undefined : formattedUpdatedAt}
        />
        <Size filesize={formattedSize} />
        <Status id={attributes.id} isAvailableOffline={isAvailableOffline} />
        <FileAction
          onClick={e => {
            onShowActionMenu(attributes.id)
            e.stopPropagation()
          }}
        />
      </div>
    )
  }
}

export default withBreakpoints()(withRouter(translate()(File)))

export const FilePlaceholder = ({ style }) => (
  <div style={style} className={styles['fil-content-row']}>
    <div
      className={classNames(
        styles['fil-content-cell'],
        styles['fil-content-file-select']
      )}
    />
    <div
      className={classNames(
        styles['fil-content-cell'],
        styles['fil-content-file']
      )}
    >
      <div className={styles['fil-content-file-placeholder']} />
    </div>
    <div
      className={classNames(
        styles['fil-content-cell'],
        styles['fil-content-date']
      )}
    />
    <div
      className={classNames(
        styles['fil-content-cell'],
        styles['fil-content-size']
      )}
    />
    <div
      className={classNames(
        styles['fil-content-cell'],
        styles['fil-content-status']
      )}
    />
    <div
      className={classNames(
        styles['fil-content-cell'],
        styles['fil-content-file-action']
      )}
    />
  </div>
)
