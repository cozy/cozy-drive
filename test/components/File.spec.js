'use strict'

import React from 'react'
import renderer from 'react-test-renderer'
import { I18n } from 'cozy-ui/transpiled/react/I18n'

import File from '../../src/components/File'

describe('File component', () => {
  it('should render a folder correctly', () => {
    const attrs = {
      id: '1234',
      name: 'Foo',
      type: 'directory',
      created_at: '2017-01-03T08:52:37.093020313Z'
    }
    const component = renderer.create(
      <I18n lang="en" dictRequire={() => ''}>
        <File attributes={attrs} />
      </I18n>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
