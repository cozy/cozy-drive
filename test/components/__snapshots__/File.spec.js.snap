// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File component should render a folder correctly 1`] = `null`;
